guides/how-nix-works.tt

```diff
@@ -11,40 +11,44 @@
 <div class="generic-layout">
   <div>
     <p>
-      Nix is a <em>purely functional package manager</em>. This means that it
-      treats packages like values in purely functional programming languages such
-      as Haskell — they are built by functions that don’t have side-effects, and
-      they never change after they have been built. Nix stores packages in the
-      <em>Nix store</em>, usually the directory <tt>/nix/store</tt>, where each
-      package has its own unique subdirectory such as
+      Nix provides <em>reproducible builds</em> and <em>correct deployment</em> of build results by ensuring <a hrefs="#complete-dependencies">complete, exact dependencies</a> both at build time and run time.
     </p>
-
-    <pre class="file-listing">/nix/store/b6gvzjyb2pg0kjfwrjmg1vfhh54ad73z-firefox-33.1/</pre>
-
     <p>
-      where <tt>b6gvzjyb2pg0…</tt> is a unique identifier for the package that
```

Comment:
I feel that by removing the only technical detail on this page (namely how Nix stores packages), we can't really call this page "How Nix works" anymore. It becomes more a listing of features, but we already have https://nixos.org/ and https://nixos.org/explore.html, so we should figure out how this page is related to those.

Reply:
> I feel that by removing the only technical detail on this page (namely how Nix stores packages), we can't really call this page "How Nix works" anymore.

Yes, the overview is supposed to be superficial. These details should go into the subsections, and I didn't want to touch them yet to keep the PR reasonable.

I'm also on the fence whether that page should instead be called "What Nix does", but that depends how we would  elaborate the subsections.

> It becomes more a listing of features, but we already have https://nixos.org/ and https://nixos.org/explore.html, so we should figure out how this page is related to those.

The front page is pretty clear IMHO, it's just the shortest possible description of what Nix can do. There still should be a complete list of major capabilities. Indeed https://nixos.org/explore.html overlap with the new introduction, and if you ask me they should be merged. Unfortunately having a design with illustrations baked in does not really allow for rearranging contents. Also, listing a disjoint set of features does not indicate how they are connected conceptually (that essentially everything falls out naturally from "complete, exact dependencies"), this is why I added that prose overview.

@garbas What does the marketing team think about this?

Reply:
While [Explore](https://nixos.org/explore.html) is to answer a question "Why would I even look into Nix?", we can assume that visitors who come here are already interested into Nix and fall into 2 different groups:
1. Already have Nix installed and use it for simple tasks (eg. simple development environments). They waited some time to really see if Nix is the tool they want to invest into. Now they want to understand how things work.
2. Before they use any tool they want to understand how it works. They might now have time to go over +20 pages (eg. nix pills), but would love to read few pages to get the basic idea how things work.

Above two groups are why this page is needed. So *How Nix/NixOS works* page can be technical. We should be able to explain in few pages basic ideas how Nix/NixOS works. We don't have to focus on *Why this is good* too much. I personally think you can explain much more (and much quicker) with a nice graphic. You don't have to design the graphic just need to provide the idea and I'll try to reach out to the designer to do the job.

Important to know about the current content of the page. Current content of the page is a copy/paste from two older pages which talked about Why you should use Nix. Ideally would would completely rewrite this page with much better explanations.

Reply:
Yes, as said I think technical details should go into subsections. I would also like to add diagrams. Right now the quickest path would be to drop in SVG files, but that is not very easy to maintain.